read-fonts/src/lib.rs

```diff
@@ -71,6 +71,86 @@ pub(crate) mod codegen_prelude {
 
 include!("../generated/font.rs");
 
+#[derive(Clone)]
+/// Reference to the content of a font or font collection file.
+pub enum FileRef<'a> {
+    /// A single font.
+    Font(FontRef<'a>),
+    /// A collection of fonts.
+    Collection(CollectionRef<'a>),
+}
+
+impl<'a> FileRef<'a> {
+    pub fn new(data: FontData<'a>) -> Result<Self, ReadError> {
+        Ok(if let Ok(collection) = CollectionRef::new(data) {
+            Self::Collection(collection)
+        } else {
+            Self::Font(FontRef::new(data)?)
+        })
+    }
+
+    pub fn fonts(&self) -> impl Iterator<Item = Result<FontRef<'a>, ReadError>> + 'a + Clone {
+        let copy = self.clone();
+        let len = match self {
+            Self::Collection(collection) => collection.len(),
+            _ => 1,
+        };
+        (0..len).filter_map(move |ix| match &copy {
+            Self::Font(font) => Some(Ok(font.clone())),
+            Self::Collection(collection) => collection.get(ix),
+        })
+    }
+}
+
+const TTC_HEADER_TAG: Tag = Tag::new(b"ttcf");
```

Comment:
Maybe this should live in `font-types`? that's where we put the consts for the sfnt versions.

We could also consider ignoring the spec in this case and treating this as a `u32`, so that it's consistent with the defined sfnt versions, which would let us do something nice, like:

```rust
// in FileRef
pub fn new(data: FontData<'a>) -> Result<Self, ReadError> {
    match data.read_at::<u32>(0)? {
        TTC_HEADER_TAG => CollectionRef::new(data).map(Self::Collection),
        TT_SFNT_VERSION | CFF_SFNT_VERSION => FontRef::new(data).map(Self::Collection),
        other => Err(ReadError::InvalidSfnt(other),
    }
}
```
We could even continue to use the `Tag` type to represent the field elsewhere, and just use a separate constant here.

I'm not sure this is particularly worth changing, but wanted to point it out.

Reply:
That was my first thought but was trying to avoid any direct parsing. I can go either way on this.